File: javascript/hamming/hamming.js
var Hamming = function() {};

Hamming.prototype.compute = function(strand1, strand2){

    if( strand1.length != strand2.length)
    {
        throw "DNA strands must be of equal length.";
    }

    var distance = 0;
    for(var i=0; i < strand1.length; i++ )
    {
        distance += strand1.charAt(i) == strand2.charAt(i) ? 0 : 1;
    }

    return distance;
};

module.exports = Hamming;

File: python/hamming/hamming.py
def distance(strand1, strand2):
    if len(strand1) != len(strand2):
        raise ValueError("The two DNA strands are not equal length")

    d = 0
    for s1, s2 in zip(strand1, strand2):
        d+= 1 if s1 != s2 else 0

    return d
